src/main-oe/sgxlkl_run_oe.c

```diff
@@ -1304,18 +1314,29 @@ void parse_cpu_affinity_params(char* config, int** cores, size_t* cores_len)
     }
 }
 
-static _Atomic(bool) first_thread = true;
-
-void* ethread_init(ethread_args_t* args)
+static void handle_ethread_exit(int exit_status)
 {
-    oe_result_t result = sgxlkl_ethread_init(args->oe_enclave);
-    bool current_value = true;
-    if (sgxlkl_config_bool(SGXLKL_VERBOSE) &&
-        atomic_compare_exchange_strong(&first_thread, &current_value, false))
+    exited_ethread_count++;
+
+    // Check if this is the terminating ethread that returns the exit
+    // status
+    if (exit_status != INT_MAX)
     {
-        sgxlkl_host_verbose("");
+        // Signal that the terminating ethread has exited
+        pthread_mutex_lock(&terminating_ethread_exited_mtx);
+        enclave_return_status = exit_status;
+        int ret = pthread_cond_signal(&terminating_ethread_exited_cv);
+        if (ret != 0)
```

Comment:
Do you expect this to fail (why are we checking for errors on the signal, but not the mutex lock / unlock)?

Reply:
Check removed.